### PR TITLE
Bg user story 50

### LINKED ISSUE #53: User Story 50, Merchant fulfills part of an order
As a merchant employee
When I visit an order show page from my dashboard
For each item of mine in the order
If the user's desired quantity is equal to or less than my current inventory quantity for that item
And I have not already "fulfilled" that item:
- Then I see a button or link to "fulfill" that item
- When I click on that link or button I am returned to the order show page
- I see the item is now fulfilled
- I also see a flash message indicating that I have fulfilled that item
- the item's inventory quantity is permanently reduced by the user's desired quantity

If I have already fulfilled this item, I see text indicating such.

### PR DESCRIPTION
**Pull request by brian-greeson**
## Merchant can fulfill orders
Closes: #53 
- Added fillable? helper method to ItemOrders model
- Added link to merchant order show page to fill each item that can be filled

### CODE REVIEW

File: app/controllers/merchant/orders_controller.rb [hunk 1]
```diff
@@ -1,13 +1,12 @@
 class Merchant::OrdersController < Merchant::BaseController
 
   def show
-    @order = Order.find(params[:id])
+    @order = Order.find(order_params[:id])
```
Comment (fredrondina) [inline app/controllers/merchant/orders_controller.rb:4]: I like the added security here

**Review by fredrondina — APPROVED** [review]
great work again brian

File: app/models/item_order.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def subtotal
   end
 
   def fulfill
-    item.update(inventory: item.inventory - quantity) if unfulfilled?
```
Comment (kelshadarby) [inline app/models/item_order.rb:12]: This is much clearer, nice fix

**Review by kelshadarby — APPROVED** [review]
Nice work Brian!

**Pull request merged**